src/agent/onefuzz-agent/src/tasks/regression/generic.rs

```diff
@@ -0,0 +1,201 @@
+                file.file_name()
+                    .to_str()
+                    .and_then(|f_name| container_url.url().join(f_name).ok())
+            });
+            let report = self
+                .get_crash_result(file.path(), input_url, self.config.check_asan_log)
+                .await?;
+
+            let reports = Some(self.config.reports.clone());
+            let no_repro = Some(self.config.no_repro.clone());
+            report
+                .save_regression(
+                    None,
+                    &reports,
+                    &no_repro,
+                    format!("{}/", self.config.common.task_id),
+                )
+                .await?;
+        }
+
+        Ok(())
+    }
+
+    pub async fn handle_crash_reports(
+        &self,
+        heartbeat_client: &Option<TaskHeartbeatClient>,
+        input_reports: &SyncedDir,
+        crashes: &SyncedDir,
+    ) -> Result<()> {
+        if self.config.report_list.is_empty() {
+            input_reports.sync_pull().await?;
+        } else {
+            for file in &self.config.report_list {
+                let input_url = input_reports
+                    .url
+                    .clone()
+                    .ok_or(format_err!("no input url"))?
+                    .blob(file);
+                download_input(input_url.url(), &input_reports.path).await?;
+            }
+        }
+
+        let mut report_files = tokio::fs::read_dir(&input_reports.path).await?;
+        while let Some(file) = report_files.next_entry().await? {
+            heartbeat_client.alive();
+            let crash_report_str = std::fs::read_to_string(file.path())?;
+            let crash_report: CrashReport = serde_json::from_str(&crash_report_str)?;
+            let input_url = crash_report
+                .input_blob
+                .clone()
+                .map(|b| b.name)
+                .and_then(|crash_name| crashes.url.clone().map(|u| u.blob(crash_name).url()));
+
+            if let Some(input_url) = input_url {
+                let input = download_input(input_url.clone(), &crashes.path).await?;
+                let report = self
+                    .get_crash_result(input, Some(input_url), crash_report.asan_log.is_some())
+                    .await?;
+                let reports = Some(self.config.reports.clone());
+                let no_repro = Some(self.config.no_repro.clone());
+                report
+                    .save_regression(
+                        Some(crash_report),
+                        &reports,
+                        &no_repro,
+                        format!("{}/", self.config.common.task_id),
+                    )
+                    .await?;
+            }
+        }
+        Ok(())
+    }
+
+    pub async fn get_crash_result(
+        &self,
+        input: impl AsRef<Path>,
+        input_url: Option<Url>,
+        is_asan: bool,
+    ) -> Result<CrashTestResult> {
+        if is_asan {
+            libfuzzer_report::test_input(
```

Comment:
The CrashReport having an `asan` entry doesn't identify if this is a libfuzzer target or not.  As an example, the radamsa template can optionally support ASAN instrumented binaries and checking for ASAN log files.

The crash reports don't have any indication as to the type of target.